text/0000-localizing-api-messages.md

```diff
@@ -0,0 +1,135 @@
+
+- Start Date: 03-29-2022
+- RFC PR: 
+- FOLIO Issue: 
+- Submitter : Radhakrishnan Gopalakrishnan (rgopalakrishnan@ebsco.com)
+- Co-Submitter(s) : Zak Burke (zburke@ebsco.com)
+- Sub Group :
+
+# Localizing API (Backend) Messages
+
+## Summary
+  The purpose of this RFC is to define an approach that allows API developer to provide
+messages in the API response depending on the language preference set in the request. 
+
+
+## Motivation
+
+- Help users understand the information coming from the application backend using a language they understand better
+- To allow FOLIO platform to be used by users across the world 
+- To create a tremendous opportunity for growth that we can never could have achieved within just one country.
+
+### Terminology
+- Static Message - E.g. _This is something you need translate_
+- Static Message with placeholder - _You can ONLY renew {n} times_
+ 
+
+### In Scope Requirements/Use cases
+- Return localized messages based on the value passed in the accept-language header
+- Handle static messages, static messages with placeholders
+
+### Out of Scope Requirements/Use cases
+- [Plural Syntax](https://wiki.folio.org/display/I18N/How+To+translate+FOLIO#HowTotranslateFOLIO-Pluralsyntax)
+- Naming convention to be used for keys
+- Process for managing (decoupled from lokalise.com) translations
```

Comment:
Respectfully, if Lokalise is needed to manage the translations, then the process for managing the translations with Lokalise is in scope.  If Lokalise is not needed, then this RFC needs to define a process by which translators will add the translated strings.

It might be possible to use the existing Lokalise→GitHub script with a few different parameters in the API call to Lokalise.  But this would need to be tested.

Reply:
I'm withdrawing this comment after watching the [TC meeting recording](https://recordings.openlibraryfoundation.org/folio/tech-council/2022-04-20T11:00/) and listening to @zburke's comments.  The use of a standard message format (like the ICU message format comment above) is even _more_ important, though, if there is a desire to decouple the architecture from the implementation mechanism.